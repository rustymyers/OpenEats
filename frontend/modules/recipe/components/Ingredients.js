import React from 'react'

export default React.createClass({
  getInitialState: function() {
    return {
      data: this.props.data || []
    };
  },

  componentWillReceiveProps(nextProps) {
    this.setState({data: nextProps.data});
  },

  render: function() {
    var ingredients = this.state.data.map(function(ingredient) {
      return (
        <li className="ingredient" key={ ingredient.id }>
          { (ingredient.quantity !== 0)
              ? <span className="quantity">{ ingredient.quantity } </span>
              : null
          }
          { (ingredient.measurement)
              ? <span className="measurement">{ ingredient.measurement } </span>
              : null
          }
          { (ingredient.title)
              ? <span className="title">{ ingredient.title }</span>
              : null
          }
        </li>
      );
    });
    return (
      <ul className="ingredients" >
        { ingredients }
      </ul>
    );
  }
});
