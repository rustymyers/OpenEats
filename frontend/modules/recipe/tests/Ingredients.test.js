import React from 'react';
import Ingredients from '../components/Ingredients';
import renderer from 'react-test-renderer';

import data from './data';

test('Ingredient component test', () => {
  const check = jest.fn();
  const component = renderer.create(
    <Ingredients data={ data.ingredients } check={ check }/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
