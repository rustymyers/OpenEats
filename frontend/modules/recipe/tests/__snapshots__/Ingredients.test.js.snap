// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ingredient component test 1`] = `
<ul
  className="ingredients"
>
  <li
    className="ingredient"
  >
    <div
      className={undefined}
    >
      <label
        className="check-container"
      >
        <input
          checked={false}
          name={1}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
    <span
      className="quantity"
    >
      1
       
    </span>
    <span
      className="measurement"
    >
      dash
       
    </span>
    <span
      className="title"
    >
      black pepper
    </span>
  </li>
  <li
    className="ingredient"
  >
    <div
      className={undefined}
    >
      <label
        className="check-container"
      >
        <input
          checked={false}
          name={2}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
    <span
      className="quantity"
    >
      4
       
    </span>
    <span
      className="measurement"
    >
      tablespoons
       
    </span>
    <span
      className="title"
    >
      chili powder
    </span>
  </li>
  <li
    className="ingredient"
  >
    <div
      className={undefined}
    >
      <label
        className="check-container"
      >
        <input
          checked={false}
          name={3}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
    <span
      className="quantity"
    >
      1
       
    </span>
    <span
      className="measurement"
    >
      tablespoon
       
    </span>
    <span
      className="title"
    >
      cumin
    </span>
  </li>
  <li
    className="ingredient"
  >
    <div
      className={undefined}
    >
      <label
        className="check-container"
      >
        <input
          checked={false}
          name={4}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
    <span
      className="quantity"
    >
      1
       
    </span>
    <span
      className="measurement"
    >
      can
       
    </span>
    <span
      className="title"
    >
      dark kidney beans
    </span>
  </li>
  <li
    className="ingredient"
  >
    <div
      className={undefined}
    >
      <label
        className="check-container"
      >
        <input
          checked={false}
          name={5}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
    <span
      className="quantity"
    >
      2
       
    </span>
    <span
      className="measurement"
    >
      cans
       
    </span>
    <span
      className="title"
    >
      diced tomatos
    </span>
  </li>
  <li
    className="ingredient"
  >
    <div
      className={undefined}
    >
      <label
        className="check-container"
      >
        <input
          checked={false}
          name={6}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
    <span
      className="quantity"
    >
      1
       
    </span>
    <span
      className="measurement"
    >
      whole
       
    </span>
    <span
      className="title"
    >
      green bell pepper
    </span>
  </li>
  <li
    className="ingredient"
  >
    <div
      className={undefined}
    >
      <label
        className="check-container"
      >
        <input
          checked={false}
          name={7}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
    <span
      className="quantity"
    >
      1
       
    </span>
    <span
      className="measurement"
    >
      pound
       
    </span>
    <span
      className="title"
    >
      ground pork
    </span>
  </li>
  <li
    className="ingredient"
  >
    <div
      className={undefined}
    >
      <label
        className="check-container"
      >
        <input
          checked={false}
          name={8}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
    <span
      className="quantity"
    >
      1
       
    </span>
    <span
      className="measurement"
    >
      pound
       
    </span>
    <span
      className="title"
    >
      ground sirloin
    </span>
  </li>
  <li
    className="ingredient"
  >
    <div
      className={undefined}
    >
      <label
        className="check-container"
      >
        <input
          checked={false}
          name={9}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
    <span
      className="quantity"
    >
      1
       
    </span>
    <span
      className="measurement"
    >
      dash
       
    </span>
    <span
      className="title"
    >
      kosher salt
    </span>
  </li>
  <li
    className="ingredient"
  >
    <div
      className={undefined}
    >
      <label
        className="check-container"
      >
        <input
          checked={false}
          name={10}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
    <span
      className="quantity"
    >
      1
       
    </span>
    <span
      className="measurement"
    >
      can
       
    </span>
    <span
      className="title"
    >
      light kidney beans
    </span>
  </li>
  <li
    className="ingredient"
  >
    <div
      className={undefined}
    >
      <label
        className="check-container"
      >
        <input
          checked={false}
          name={11}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
    <span
      className="quantity"
    >
      1
       
    </span>
    <span
      className="measurement"
    >
      whole
       
    </span>
    <span
      className="title"
    >
      serrano pepper 
    </span>
  </li>
  <li
    className="ingredient"
  >
    <div
      className={undefined}
    >
      <label
        className="check-container"
      >
        <input
          checked={false}
          name={12}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
    <span
      className="quantity"
    >
      1
       
    </span>
    <span
      className="measurement"
    >
      whole
       
    </span>
    <span
      className="title"
    >
      white onion
    </span>
  </li>
</ul>
`;
